/tt-metalium/include/ttnn/operations/conv/conv2d/conv2d.hpp missing

### Describe the bug

Hi, I am trying to compile the output of  codegen_cpp( ) but it is complaining about missing files from my tt-metalium installation. I built from sources with:

git clone https://github.com/tenstorrent/tt-metal.git --recurse-submodules
cmake --build
 cmake --install .

After installation I get:

rbeattie@tt-devel:~$ ls tt-metalium-local/include/ttnn/operations/
cb_utils.hpp  compute_throttle_utils.hpp  core          data_movement  embedding           experimental  functions.hpp  math.hpp  moreh          rand       trace.hpp
ccl           copy                        creation.hpp  eltwise        embedding_backward  full_like     kv_cache       matmul    normalization  reduction

There are no convolutions so when I try to build the codegen output I get:

[ 50%] Building CXX object CMakeFiles/custom.dir/main.cpp.o
In file included from /home/rbeattie/share/examples/cxx/custom/main.cpp:1:
/home/rbeattie/share/examples/cxx/custom/header.hpp:13:10: fatal error: ttnn/operations/conv/conv2d/conv2d.hpp: No such file or directory
   13 | #include "ttnn/operations/conv/conv2d/conv2d.hpp"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~




### Steps to reproduce the issue

1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error


### Expected behavior

_No response_

### Please complete the following environment information

_No response_

--- a/ttnn/cpp/ttnn/operations/conv/CMakeLists.txt
+++ b/ttnn/cpp/ttnn/operations/conv/CMakeLists.txt
@@ -19,6 +19,11 @@ target_sources(
         TYPE HEADERS
         BASE_DIRS ${FixmeOpAPIDir}
         FILES
+            conv2d/conv2d.hpp
+            conv2d/device/conv2d_device_operation_types.hpp
+            conv2d/device/conv2d_device_operation.hpp
+            conv2d/device/conv2d_op_sharded_program_factory.hpp
+            conv2d/device/conv2d_op_width_sharded_program_factory.hpp
         FILE_SET kernels
         TYPE HEADERS
         BASE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}
@@ -52,4 +57,4 @@ install(
         DESTINATION ${CMAKE_INSTALL_LIBEXECDIR}/tt-metalium/ttnn/cpp/ttnn/operations/conv
         COMPONENT ttnn-runtime
 )
-install(TARGETS ttnn_op_conv LIBRARY COMPONENT tar)
+install(TARGETS ttnn_op_conv FILE_SET api COMPONENT ttnn-dev LIBRARY COMPONENT tar)

--- a/ttnn/cpp/ttnn/operations/conv/CMakeLists.txt
+++ b/ttnn/cpp/ttnn/operations/conv/CMakeLists.txt
@@ -42,12 +42,7 @@ target_sources(
 )
 
 target_include_directories(ttnn_op_conv PRIVATE ${FixmeOpIncDirs})
-target_link_libraries(
-    ttnn_op_conv
-    PRIVATE
-        TT::Metalium
-        TTNN::Core
-)
+target_link_libraries(ttnn_op_conv PUBLIC TTNN::Core PRIVATE TT::Metalium)
 
 install(
     TARGETS

--- a/ttnn/cpp/ttnn/operations/eltwise/unary/CMakeLists.txt
+++ b/ttnn/cpp/ttnn/operations/eltwise/unary/CMakeLists.txt
@@ -17,7 +17,12 @@ target_sources(
         FILE_SET api
         TYPE HEADERS
         BASE_DIRS ${FixmeOpAPIDir}
-        FILES common/unary_op_types.hpp device/unary_composite_op.hpp unary.hpp unary_composite.hpp
+        FILES
+            common/unary_op_types.hpp
+            common/unary_op_utils.hpp
+            device/unary_composite_op.hpp
+            unary.hpp
+            unary_composite.hpp
         FILE_SET kernels
         TYPE HEADERS
         BASE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}

--- a/ttnn/cpp/ttnn/operations/sliding_window/CMakeLists.txt
+++ b/ttnn/cpp/ttnn/operations/sliding_window/CMakeLists.txt
@@ -18,7 +18,7 @@ target_sources(
         FILE_SET api
         TYPE HEADERS
         BASE_DIRS ${FixmeOpAPIDir}
-        FILES
+        FILES sliding_window.hpp op_slicing/op_slicing.hpp
         FILE_SET kernels
         TYPE HEADERS
         BASE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}
@@ -47,4 +47,4 @@ install(
         DESTINATION ${CMAKE_INSTALL_LIBEXECDIR}/tt-metalium/ttnn/cpp/ttnn/operations/sliding_window
         COMPONENT ttnn-runtime
 )
-install(TARGETS ttnn_op_sliding_window LIBRARY COMPONENT tar)
+install(TARGETS ttnn_op_sliding_window FILE_SET api COMPONENT ttnn-dev LIBRARY COMPONENT tar)

--- a/ttnn/cpp/ttnn/operations/sliding_window/CMakeLists.txt
+++ b/ttnn/cpp/ttnn/operations/sliding_window/CMakeLists.txt
@@ -32,12 +32,7 @@ target_sources(
 )
 
 target_include_directories(ttnn_op_sliding_window PRIVATE ${FixmeOpIncDirs})
-target_link_libraries(
-    ttnn_op_sliding_window
-    PRIVATE
-        TT::Metalium
-        TTNN::Core
-)
+target_link_libraries(ttnn_op_sliding_window PUBLIC TTNN::Core PRIVATE TT::Metalium)
 
 install(
     TARGETS

